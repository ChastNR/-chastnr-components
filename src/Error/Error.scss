.er__cntr {
  transition: height 0.25s ease;
  overflow: hidden;

  & > div {
    color: #e31c1c;
    margin-bottom: -1px;
    opacity: 0;
    padding-top: 5px;
    text-align: left;
    transition-delay: 0.1s;
    transition: opacity 0.25s;
  }

  &.error {
    & > div {
      opacity: 1;
    }
  }
}
