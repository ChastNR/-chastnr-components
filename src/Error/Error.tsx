import { useState } from "react";

import classNames from "clsx";

import "./Error.scss";

const Error: React.FC = ({ children }) => {
  const [height, setHeight] = useState(0);

  const handleRef = (element: HTMLDivElement | null) =>
    element && setHeight(children ? element.scrollHeight : 0);

  const errorClassNames = classNames("er__cntr", {
    error: height !== 0,
  });

  return (
    <div className={errorClassNames} ref={handleRef} style={{ height }}>
      <div>{children}</div>
    </div>
  );
};

export default Error;
