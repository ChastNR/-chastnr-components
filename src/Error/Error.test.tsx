import { render } from '@testing-library/react';
import { Error } from './Error';

describe('<Error/>', () => {
  it('should match snapshot', () => {
    const error = render(<Error>Error Text</Error>);
    expect(error.container.firstChild).toMatchSnapshot();
  });
});
