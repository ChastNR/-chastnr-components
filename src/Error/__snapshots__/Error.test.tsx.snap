// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error/> should match snapshot 1`] = `
<div
  class="er__cntr"
  style="height: 0px;"
>
  <div>
    Error Text
  </div>
</div>
`;
